airbyte-integrations/connectors/source-posthog/source_posthog/source.py

```diff
@@ -0,0 +1,81 @@
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from typing import Any, List, Mapping, Tuple
+
+import pendulum
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+from .streams import (  # EventsSessions,
+    Annotations,
+    Cohorts,
+    Elements,
+    Events,
+    FeatureFlags,
+    Insights,
+    InsightsPath,
+    InsightsSessions,
+    Persons,
+    Trends,
+)
+
+
+class SourcePosthog(AbstractSource):
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
+        try:
+            _ = pendulum.parse(config["start_date"], strict=True)
+            authenticator = TokenAuthenticator(token=config["api_key"])
+            annotations_stream = Annotations(authenticator=authenticator)
+            records = annotations_stream.read_records(sync_mode=SyncMode.full_refresh)
+            _ = next(records)
+            return True, None
+        except Exception as e:
+            return False, repr(e)
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        """
+        event/sessions stream is dynamic. Probably, it contains a list of CURRENT sessions.
+        In Next day session may expire and wont be available via this endpoint.
+        So we need a dynamic load data before tests.
+        This stream was requested to be removed due to this reason.
+        """
+        authenticator = TokenAuthenticator(token=config["api_key"])
+        return [
+            Annotations(authenticator=authenticator),
+            Cohorts(authenticator=authenticator),
+            Elements(authenticator=authenticator),
+            Events(start_date=config["start_date"], authenticator=authenticator),
+            # EventsSessions(authenticator=authenticator),
```

Comment:
Why should we remove it?
Perhaps the user will launch this stream once an hour and receive data about user sessions.

Reply:
this is a dynamic stream, it stores current sessions. Today we can have the data, tomorrow it will have 0 records in the response (get this behaviour 2 days in a row). The user may have empty data in it. Also, we have to add the data right before tests to pass data-existense in the endpoint.

Reply:
But the user can just put the "Append" method and get even sometimes repeated records, but new ones too?
And for tests, we can skip this test in this case.